### PR TITLE
Use initrd.target in the initramfs

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/core/main.c [hunk 1]
```diff
@@ -1967,20 +1967,33 @@ static int do_queue_default_job(
...
+                default_unit = SPECIAL_INITRD_TARGET;
+        else
+                default_unit = SPECIAL_DEFAULT_TARGET;
+
+        log_debug("Activating default unit: %s", default_unit);
 
-        r = manager_load_startable_unit_or_warn(m, arg_default_unit, NULL, &target);
+        r = manager_load_startable_unit_or_warn(m, default_unit, NULL, &target);
+        if (r < 0 && in_initrd()) {
+                log_info("Falling back to " SPECIAL_INITRD_TARGET ".");
```
Comment (poettering) [inline src/core/main.c]: hmm, we'd typically try initrd.target twice with this, no? doesn't look right?

**Comment by poettering** [comment]
(i presume this is stuff intended for v245)

File: src/core/main.c (hunk 1 above)
Comment (keszybz) [reply src/core/main.c]: Argh, I made the fall back to the wrong target. Should be fixed now.

**Comment by keszybz** [comment]
> (i presume this is stuff intended for v245)

Yep.

**Comment by keszybz** [comment]
I pushed one more commit: make sure that systemd.debug-shell works in the initramfs. It is simple and similar in substance to previous commits, so I'm keeping the review flag.

Unfortunately dracut does not include `systemd-debug-generator` in the initramfs, so at least for Fedora users this change is moot.

**Comment by lnykryn** [comment]
> Unfortunately dracut does not include `systemd-debug-generator` in the initramfs, so at least for Fedora users this change is moot.

Btw this should be fixed in next dracut version:
https://github.com/dracutdevs/dracut/commit/707d4e79feadb475b79fd8f15dd2fea70aa79869

**Pull request merged**